# fix up libsysfs header file usage to fix bug reports from users that have sysfsutils installed already.

diff -Nru a/libsysfs/sysfs/libsysfs.h b/libsysfs/sysfs/libsysfs.h
--- a/libsysfs/sysfs/libsysfs.h	Mon Feb 23 16:35:20 2004
+++ b/libsysfs/sysfs/libsysfs.h	Mon Feb 23 16:35:20 2004
@@ -24,7 +24,6 @@
 #define _LIBSYSFS_H_
 
 #include <sys/types.h>
-#include "dlist.h"
 
 /*
  * Generic #defines go here..
@@ -46,6 +45,8 @@
 
 #define SYSFS_METHOD_SHOW	0x01	/* attr can be read by user */
 #define SYSFS_METHOD_STORE	0x02	/* attr can be changed by user */
+
+struct dlist;
 
 struct sysfs_attribute {
 	unsigned char *value;
diff -Nru a/libsysfs/sysfs.h b/libsysfs/sysfs.h
--- a/libsysfs/sysfs.h	Mon Feb 23 16:35:20 2004
+++ b/libsysfs/sysfs.h	Mon Feb 23 16:35:20 2004
@@ -31,6 +31,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
+#include "dlist.h"
 
 /* Debugging */
 #ifdef DEBUG
diff -Nru a/namedev.c b/namedev.c
--- a/namedev.c	Mon Feb 23 16:35:24 2004
+++ b/namedev.c	Mon Feb 23 16:35:24 2004
@@ -32,8 +32,8 @@
 #include <time.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
-#include <sysfs/libsysfs.h>
 
+#include "libsysfs/sysfs/libsysfs.h"
 #include "list.h"
 #include "udev.h"
 #include "udev_version.h"
diff -Nru a/udev-add.c b/udev-add.c
--- a/udev-add.c	Mon Feb 23 16:35:24 2004
+++ b/udev-add.c	Mon Feb 23 16:35:24 2004
@@ -33,8 +33,8 @@
 #ifndef __KLIBC__
 #include <pwd.h>
 #endif
-#include <sysfs/libsysfs.h>
 
+#include "libsysfs/sysfs/libsysfs.h"
 #include "udev.h"
 #include "udev_version.h"
 #include "udev_dbus.h"
diff -Nru a/udev.c b/udev.c
--- a/udev.c	Mon Feb 23 16:35:24 2004
+++ b/udev.c	Mon Feb 23 16:35:24 2004
@@ -26,8 +26,8 @@
 #include <errno.h>
 #include <ctype.h>
 #include <signal.h>
-#include <sysfs/libsysfs.h>
 
+#include "libsysfs/sysfs/libsysfs.h"
 #include "udev.h"
 #include "udev_version.h"
 #include "udev_dbus.h"
diff -Nru a/udev_config.c b/udev_config.c
--- a/udev_config.c	Mon Feb 23 16:35:24 2004
+++ b/udev_config.c	Mon Feb 23 16:35:24 2004
@@ -31,8 +31,8 @@
 #include <unistd.h>
 #include <errno.h>
 #include <ctype.h>
-#include <sysfs/libsysfs.h>
 
+#include "libsysfs/sysfs/libsysfs.h"
 #include "udev.h"
 #include "udev_version.h"
 #include "logging.h"
diff -Nru a/udevdb.c b/udevdb.c
--- a/udevdb.c	Mon Feb 23 16:35:24 2004
+++ b/udevdb.c	Mon Feb 23 16:35:24 2004
@@ -32,8 +32,8 @@
 #include <sys/stat.h>
 #include <errno.h>
 #include <signal.h>
-#include <sysfs/libsysfs.h>
 
+#include "libsysfs/sysfs/libsysfs.h"
 #include "udev_version.h"
 #include "udev.h"
 #include "logging.h"
diff -Nru a/udevinfo.c b/udevinfo.c
--- a/udevinfo.c	Mon Feb 23 16:35:24 2004
+++ b/udevinfo.c	Mon Feb 23 16:35:24 2004
@@ -26,8 +26,9 @@
 #include <stdarg.h>
 #include <unistd.h>
 #include <errno.h>
-#include <sysfs/libsysfs.h>
 
+#include "libsysfs/sysfs/libsysfs.h"
+#include "libsysfs/dlist.h"
 #include "udev.h"
 #include "udev_version.h"
 #include "logging.h"
diff -Nru a/udevtest.c b/udevtest.c
--- a/udevtest.c	Mon Feb 23 16:35:24 2004
+++ b/udevtest.c	Mon Feb 23 16:35:24 2004
@@ -26,8 +26,8 @@
 #include <errno.h>
 #include <ctype.h>
 #include <signal.h>
-#include <sysfs/libsysfs.h>
 
+#include "libsysfs/sysfs/libsysfs.h"
 #include "udev.h"
 #include "udev_version.h"
 #include "logging.h"
